### PR TITLE
Update the auth event manager to handle redirects. Add redirect logic to popupredirectresolver

### PR DESCRIPTION
**Pull request by samhorlbeck**

### CODE REVIEW

File: packages-exp/auth-exp/src/core/auth/auth_event_manager.ts [hunk 1]
```diff
@@ -18,43 +18,108 @@
...
+      this.queuedRedirectEvent = null;
+    }
   }
 
   unregisterConsumer(authEventConsumer: AuthEventConsumer): void {
     this.consumers.delete(authEventConsumer);
   }
 
-  onEvent(event: AuthEvent): void {
+  onEvent(event: AuthEvent): boolean {
```
Comment (avolkovi) [inline packages-exp/auth-exp/src/core/auth/auth_event_manager.ts:49]: this is ok... potentially worth refactoring eventually to handle redirect events in a completely separate path from popup events

File: packages-exp/auth-exp/src/core/auth/auth_event_manager.ts [hunk 2]
```diff
@@ -18,43 +18,108 @@
...
+      this.queuedRedirectEvent = event;
+      handled = true;
+    }
+
+    return handled;
+  }
+
+  private sendToConsumer(event: AuthEvent, consumer: AuthEventConsumer): void {
+    if (event.error) {
+      console.error('ERROR');
```
Comment (avolkovi) [inline packages-exp/auth-exp/src/core/auth/auth_event_manager.ts:84]: stray console()

File: packages-exp/auth-exp/src/core/strategies/popup.ts [hunk 3]
```diff
@@ -118,7 +118,7 @@ class PopupOperation extends AbstractPopupRedirectOperation {
     this.authWindow = await this.resolver._openPopup(
       this.auth,
       this.provider,
-      this.filter,
+      this.filter[0], // There's always one, see constructor
```
Comment (avolkovi) [inline packages-exp/auth-exp/src/core/strategies/popup.ts:121]: add an assert?

File: packages-exp/auth-exp/demo/src/index.js [hunk 4]
```diff
@@ -50,26 +50,17 @@ import {
   signInWithCustomToken,
   signInWithEmailAndPassword,
   unlink,
-<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (samhorlbeck) [reply packages-exp/auth-exp/demo/src/index.js:53]: yeah no idea

File: packages-exp/auth-exp/src/core/auth/auth_event_manager.ts (hunk 1 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/auth/auth_event_manager.ts:49]: Agreed. Added a comment in popup_redirect.ts

File: packages-exp/auth-exp/src/core/auth/auth_event_manager.ts (hunk 2 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/auth/auth_event_manager.ts]: Done

File: packages-exp/auth-exp/src/core/strategies/popup.ts (hunk 3 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/strategies/popup.ts:126]: Done

**Pull request merged**